jaeger-core/src/test/java/io/jaegertracing/internal/JaegerSubclassTest.java

```diff
@@ -0,0 +1,179 @@
+package io.jaegertracing.internal;
+
+import io.jaegertracing.internal.clock.Clock;
+import io.jaegertracing.internal.metrics.Metrics;
+import io.jaegertracing.spi.BaggageRestrictionManager;
+import io.jaegertracing.spi.Reporter;
+import io.jaegertracing.spi.Sampler;
+import io.opentracing.Scope;
+import io.opentracing.ScopeManager;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class JaegerSubclassTest {
+  private static class Configuration extends io.jaegertracing.Configuration {
+    Configuration(io.jaegertracing.Configuration config) {
+      this(config.getServiceName());
+      withCodec(config.getCodec());
+      withMetricsFactory(config.getMetricsFactory());
+      withReporter(config.getReporter());
+      withSampler(config.getSampler());
+      withTracerTags(config.getTracerTags());
+    }
+
+    public Configuration(String serviceName) {
+      super(serviceName);
+    }
+
+    public static Configuration fromEnv() {
+      return new Configuration(io.jaegertracing.Configuration.fromEnv());
+    }
+
+    @Override
+    public synchronized Tracer getTracer() {
+      return new Tracer(super.getTracer());
+    }
+  }
+
+  private static class Tracer extends JaegerTracer {
+    public class SpanBuilder extends JaegerTracer.SpanBuilder {
+      protected SpanBuilder(String operationName) {
+        super(operationName);
+      }
+
+      @Override
+      public Span start() {
+        return new Span(super.start());
+      }
+
+      @Override
+      public Scope startActive(boolean finishSpanOnClose) {
+        return scopeManager().activate(start(), finishSpanOnClose);
+      }
+    }
+
+    Tracer(io.jaegertracing.internal.JaegerTracer tracer) {
```

Comment:
You probably don't need the fully qualified class name here, do you?

Reply:
Good catch.